### PR TITLE
[Bugfix] Honor attention logit softcapping in paged SDPA

### LINKED ISSUE #665: [Bug] attn_logit_softcapping is hardcoded to 0.0 in the paged SDPA kernel call
## Summary

`sdpa_forward` passes the attention softcap to the Metal kernel as a hardcoded constant, at both call sites:

```python
            cache_kv_heads,
            attn_scale,
            0.0,  # softcap (0 = disabled)
            block_tables,
```

The kernel already accepts a softcap parameter — vllm-metal simply never populates it. Architectures that set `attn_logit_softcapping` therefore run without it. Gemma 2 sets `attn_logit_softcapping: 50.0` and `mlx_lm/models/gemma2.py` applies it unconditionally:

```python
scores = mx.tanh(scores / self.attn_logit_softcapping)
scores *= self.attn_logit_softcapping
```

The output-side cap is handled correctly — `_apply_target_logit_postprocessing` reads `final_logit_softcapping` (30.0) and applies it. Only the attention-side cap is dropped.

Gemma 3 and Gemma 4 do not use attention softcapping (they use QK norm), which is presumably why the hardcoded `0.0` has gone unnoticed.

## Confirmed behaviorally, not just by reading source

Greedy, chat-templated, 24 tokens, `mlx-community/gemma-2-2b-it-4bit` @ `2c715097ff9c`, identical prompt strings on both sides.

I patched `mlx_lm`'s Gemma 2 attention to skip **only** the two softcap lines and compared three ways:

| Comparison | Result |
| --- | --- |
| stock vs stock (determinism control) | identical |
| stock vs my re-implementation *with* softcap (harness control) | identical |
| **`mlx_lm` softcap-disabled vs vllm-metal paged** | **identical, 24/24** |
| stock `mlx_lm` vs vllm-metal paged | diverges at token 17 |

On the prompt `"What is the capital of France?"` the paged path reproduces softcap-disabled `mlx_lm` exactly, and differs from stock. That is the defect, isolated.

## Impact today: real but near-inert at 2B

Worth stating plainly so this is prioritised correctly. I instrumented the actual pre-softcap attention scores over all 26 layers on a real prompt:

```
attn_logit_softcapping = 50.0
pre-softcap |scores|:  max = 16.16   mean = 10.11   min = 6.34
layers whose max |score| exceeds the cap (50.0):   0 / 26
layers exceeding even half the cap (25.0):         0 / 26
```

`tanh(x/50)*50` is close to identity for |x| ≤ 16, so the cap never binds on this checkpoint. The observable effect was confined to a trailing padding token (index 17), after the answer text had ended, with `ignore_eos=True` forcing generation past EOS.

**Unmeasured at larger scales.** Attention logits grow with model width and context length, and I could not test `gemma-2-9b` or `gemma-2-27b` (8 GB machine). If scores approach 50 there, the missing cap stops being near-inert. Flagging as future risk rather than a claim.

## Suggested fix

Plumb the value instead of hardcoding it — read `attn_logit_softcapping` off the attention module (or its args) and pass it through `sdpa_forward` to the kernel, defaulting to `0.0` when absent so every current architecture is unaffected.

## Separate observation, not part of this report

On one other prompt (`"Explain in two sentences why the sky appears blue."`) the paged path diverges from `mlx_lm` at token 4, and softcapping does **not** explain it — with softcap disabled in `mlx_lm` the output is unchanged there, yet paged still differs. The decision is a 0.078-logit margin between ` because` (20.484) and ` due` (20.406), so any small numerical difference flips it. Single data point, cause not isolated. Recording it here for completeness only — I am not asserting it as a bug and have not opened an issue for it.

## Note on the supported-models table

Not adding a Gemma 2 row, per this table's own guidance that a model which does not work should get an issue rather than a row. Two of four prompts diverged from the `mlx_lm` reference, so a clean row would misrepresent the state.

## Environment

Apple M2 / 8 GB, macOS 26.6.2, Python 3.12.2 (arm64), vllm 0.28.0+cpu, mlx 0.32.0, mlx-lm 0.31.3, vllm-metal at `main` @ `f61e7a6`. Checkpoint `mlx-community/gemma-2-2b-it-4bit` @ `2c715097ff9c081a6ac1e5cd239e2ac756b5bd99` (`Gemma2ForCausalLM`, 26 layers, 8/4 heads, head_dim 256).

### PR DESCRIPTION
**Pull request by HanningLin**
## Summary

- read the existing `attn_logit_softcapping` value from each wrapped attention module
- pass it to both the dense and TurboQuant paged-attention kernel calls
- preserve the current disabled fast path (`0.0`) when the attribute is absent or `None`
- add regression coverage for Gemma 2-style `50.0` propagation and the disabled fallback

No new model or wrapper attribute is introduced.

Fixes #665

## Testing

- `pytest -q tests/test_attention_sdpa.py tests/test_model_adapter.py` (103 passed)
- `ruff check vllm_metal/attention/impls/sdpa.py tests/test_attention_sdpa.py`
- `ruff format --check vllm_metal/attention/impls/sdpa.py tests/test_attention_sdpa.py`
- `git diff --check`

Broader native-kernel tests were not usable in this checkout because its prebuilt Metal artifacts are stale relative to the source timestamps. This change does not modify Metal or C++ sources.

### CODE REVIEW

**Comment by HanningLin** [comment]
Duplicate to https://github.com/vllm-project/vllm-metal/pull/666
